app/controllers/api/v1/users_controller.rb

```diff
@@ -29,11 +31,36 @@ def index
   param :page, :number, required: false, desc: 'Page in results (25 per page)'
   error 404, "User not found"
   def posts
-    return unless (user = find_object(User))
-
-    post_ids = Post::Author.where(user: user).pluck(:post_id)
+    post_ids = Post::Author.where(user: @user).pluck(:post_id)
     queryset = Post.privacy_public.where(id: post_ids).with_reply_count.select('posts.*')
     posts = paginate queryset.includes(:board, :joined_authors, :section), per_page: 25
     render json: { results: posts }
   end
+
+  api :GET, '/users/:id/bookmarks', "Load all of a user's bookmarks, optionally limited to a single post"
+  param :id, :number, required: true, desc: "User ID"
+  param :post_id, :number, required: false, desc: "Post ID"
+  param :page, :number, required: false, desc: 'Page in results (25 per page)'
+  error 403, "Bookmarks are not visible to the user"
+  error 404, "User not found"
+  error 422, "Invalid parameters provided"
+  def bookmarks
+    if !@user.public_bookmarks && @user.id != current_user.try(:id)
+      error = { message: "This user's bookmarks are private." }
+      render json: { errors: [error] }, status: :forbidden
+      return
+    end
+
+    bookmarks = @user.bookmarks.visible_to(current_user)
+    bookmarks = bookmarks.where(post_id: params[:post_id]) if params[:post_id].present?
+
+    bookmarks = paginate bookmarks, per_page: 25
```

Comment:
Would be a good idea to order these, since we're paginating them! I don't think we have a default order here, do we?

```suggestion
    bookmarks = paginate bookmarks.order(:id), per_page: 25
```